File: docker/Dockerfile.java
# Vulnerable Dockerfile - Java Application
# DO NOT USE IN PRODUCTION!

# VULNERABILITY: Using OpenJDK 11 without specific digest
FROM openjdk:11-jdk-slim

# VULNERABILITY: Running as root
WORKDIR /app

# VULNERABILITY: Unpinned apt packages
RUN apt-get update && apt-get install -y \
    curl \
    wget \
    git \
    maven \
    && rm -rf /var/lib/apt/lists/*

# VULNERABILITY: Maven wrapper not verified
RUN curl -L https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.1.0/maven-wrapper-3.1.0.jar -o maven-wrapper.jar

COPY pom.xml .
COPY src ./src

# VULNERABILITY: Building with potentially malicious dependencies
RUN mvn package -DskipTests

# VULNERABILITY: Running with debug enabled
ENV JAVA_OPTS="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=*:5005"

# VULNERABILITY: Multiple ports exposed
EXPOSE 8080 5005

# VULNERABILITY: Running as root with debug port exposed
CMD ["java", "-jar", "target/vuln-java-app-1.0.0.jar"]

File: docker/Dockerfile.go
# Vulnerable Dockerfile - Go Application  
# DO NOT USE IN PRODUCTION!

# VULNERABILITY: Using older Go version with known CVEs
FROM golang:1.18-alpine

# VULNERABILITY: Running as root
WORKDIR /app

# VULNERABILITY: Unpinned packages, no version locks
RUN apk add --no-cache \
    git \
    gcc \
    musl-dev \
    curl \
    openssh

# VULNERABILITY: Git config allows arbitrary code execution
RUN git config --global url."https://".insteadOf git://

# Copy vulnerable go.mod and go.sum
COPY go.mod go.sum ./

# VULNERABILITY: Downloading dependencies without verification
RUN go mod download

COPY . .

# VULNERABILITY: Building without security flags
RUN go build -o main .

# VULNERABILITY: No scratch/distroless final stage
# VULNERABILITY: Contains build tools in production image

EXPOSE 8080

CMD ["./main"]
